Add monitoring to individual functions in sync

aidenkeating opened the pull request:
This adds time monitoring to many of the functions in fh-mbaas-api.

The monitoring sorted each record by `module` and usually by method
name or by a description in the value for the `method` key of the
record.

I don't know if it makes sense to merge this into the current `monitoring` branch or not. Just adding a PR here for review purposes.

@david-martin @wtrocki Could you take a look please?

Commit:
Add monitoring to individual functions in sync

This adds time monitoring to many of the functions in fh-mbaas-api.

The monitoring sorted each record by `module` and usually by method
name or by a description in the value for the `method` key of the
record.

david-martin commented on lib/sync-DataSetModel.js:
```diff
@@ -350,13 +404,21 @@ var self = {
     var metaDataHash = syncUtil.generateHash(meta_data);
 
+    var timeTaken = Date.now() - startTime;
+    self.apiMetrics.gauge()(self.apiMetrics + '_api_timing',{method:"doSyncLoop"}, timeTaken);
```
getClientHash?

david-martin commented on lib/sync-DataSetModel.js:
```diff
@@ -564,10 +629,15 @@ var self = {
 };
 
 var init = function (apiMetrics) {
+
```
could we call defaultDataHandler.setAPIMetrics here instead of changing the signature?

wtrocki commented:
Awesome change. Please note that this may affect api timing so you cannot relay on api metrics from previous runs. With such amount of different metrics we should probably change way we display those in graphana. We should probably group by time and use max to show what parts of the code are taking most of the time.

david-martin commented:
@wtrocki Thanks Wojciech.
We're updating grafana dashboard in https://issues.jboss.org/browse/FH-3074 to have a good view of the most important fns and their execution times during load testing

Commit:
Update function timing metrics

  * Update requirement of apiMetrics to not be so obtrusive.

  * Include sync-util metrics

david-martin commented:
:+1:

Commit:
Comment out remote datahandler metrics calls

Commit:
Update fh-component-metrics version to 2.3.0

Commit:
Remove unneeded stuff

Commit:
Add dataset client count

Commit:
FH-3105 Respond immediately when changes are received.

This moves the `returnUpdates` invocation to almost immediately
after the request has started.

It also removes some unneeded code around which hash to append to
the `res` object before calling `returnUpdates`.

Commit:
FH-3105 Remove comments

Removes comments around code and reasons why code was removed for
this change.

Commit:
Update fh-component-metrics dependency

wtrocki commented:
Closing as oudated
